### PR TITLE
Add keyboard shorcuts again, now in the global section

### LINKED ISSUE #978: Question: check if notifications are visible
I'm using Sway and am trying to map my `Escape` key to:

- If any notification is visible: close all notifications via `dunstctl`
- If none are visible: re-send `Escape` key via `wtype`

This way I can use my `Escape` key to be able to close the notifications and still use it as-is.

Is it possible to somehow check if any notifications are currently being displayed by dunst?

### PR DESCRIPTION
**Pull request by fwsmit**
fixes #978

### CODE REVIEW

**Comment by fwsmit** [comment]
@wgmayer0, could you test this PR. The shortcut settings have been moved to `[global]` for consistency with other settings.

**Comment by wgmayer0** [comment]
> @wgmayer0, could you test this PR. The shortcut settings have been moved to `[global]` for consistency with other settings.

Sorry I am on Arch Linux and I am new to open source. I thought the Arch Linux maintainer was going to update the package and then I was going to test it. Could you show me a guide that tells me the steps to testing this on Arch? Or is anyone else able to help test that is already competent in the testing process?

File: src/settings_data.h [hunk 1]
```diff
@@ -1412,6 +1412,47 @@ static const struct setting allowed_settings[] = {
                 .parser_data = NULL,
         },
 
+        // Keybaord shortcuts (still in global section)
```
Comment (wgmayer0) [inline src/settings_data.h:1415]: check keyboard spelling

**Comment by fwsmit** [comment]
> Sorry I am on Arch Linux and I am new to open source. I thought the Arch Linux maintainer was going to update the package and then I was going to test it. Could you show me a guide that tells me the steps to testing this on Arch? Or is anyone else able to help test that is already competent in the testing process?

Here are the steps for running dunst from source, it isn't hard:

If you haven't already, install the [base-devel](https://archlinux.org/groups/x86_64/base-devel/) package, and the packages listsed in the dependencies row of https://aur.archlinux.org/packages/dunst-git.

Make sure to move your settings in the `[shortcuts]` section to the `[global]` section. 

Then to build and run:
```
git clone https://github.com/dunst-project/dunst.git
cd dunst
make
pkill dunst
./dunst
```

**Comment by wgmayer0** [comment]
Thanks I got it downloaded and installed. I don't have my code anymore for the keybinding I use which is just ESCAPE key mapped to the close button. Can you give me sample config code for my config and I will test it?

**Comment by fwsmit** [comment]
> Thanks I got it downloaded and installed. I don't have my code anymore for the keybinding I use which is just ESCAPE key mapped to the close button. Can you give me sample config code for my config and I will test it?

I'm not sure what the name of escape is, but I'm guessing escape. Let me know if it works. Also, please try out the other shortcuts. The context menu can be tested out with `dunstify -A "default,default" "test"`. Make sure to have dmenu installed.

```ini
[global]
    # Shortcuts are specified as [modifier+][modifier+]...key
    # Available modifiers are "ctrl", "mod1" (the alt-key), "mod2",
    # "mod3" and "mod4" (windows-key).
    # Xev might be helpful to find names for keys.

    # Close notification. Equivalent dunstctl command:
    # dunstctl close
    #close = ctrl+space
    close = escape
   

    # Close all notifications. Equivalent dunstctl command:
    # dunstctl close-all
    close_all = ctrl+shift+space

    # Redisplay last message(s). Equivalent dunstctl command:
    # dunstctl history-pop
    history = ctrl+grave

    # Context menu. Equivalent dunstctl command:
    # dunstctl context
    context = ctrl+shift+period
```

**Comment by wgmayer0** [comment]
So I tried putting

[global]

    # Close notification. Equivalent dunstctl command:
    # dunstctl close
    close = ctrl+space
    #close = escape

    # Close all notifications. Equivalent dunstctl command:
    # dunstctl close-all
    close_all = ctrl+shift+space

    # Redisplay last message(s). Equivalent dunstctl command:
    # dunstctl history-pop
    history = ctrl+grave

    # Context menu. Equivalent dunstctl command:
    # dunstctl context
    context = ctrl+shift+period

    ### Display ###

    # Which monitor should the notifications be displayed on.
    ...

in /home/wgm/dunst/dunstrc and also in /home/wgm/.config/dunst/dunstrc and I still was not able to get any of the shortcuts to work even though I uninstalled the official release from Arch's package manager.

**Comment by fwsmit** [comment]
>in /home/wgm/dunst/dunstrc 
This one is not read. Feel free to delete it.

> and also in /home/wgm/.config/dunst/dunstrc

Yes, this is the correct location.

> I still was not able to get any of the shortcuts to work even though I uninstalled the official release from Arch's package manager.

I'm sorry I didn't give the correct instructions to run this PR's code (since it's not merged yet). The easiest way to run it is to install the github command line tool and then compile and run the code.

So when inside the cloned repository (after the `cd dunst` step) do this instead:

```
sudo pacman -S github-cli
gh pr checkout 1033
make
pkill dunst
./dunst
```

P.S.

You can create code blocks like the above by surrounding it in backticks (\`, key under the escape key). See also https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

**Comment by wgmayer0** [comment]
The shortcuts work, but the escape key itself does not. I know a lot of people like to use escape.

**Comment by wgmayer0** [comment]
I am also seeing "WARNING: Unknown keyboard shortcut: 'escape'"

**Comment by fwsmit** [comment]
Thanks for testing!

> I am also seeing "WARNING: Unknown keyboard shortcut: 'escape'"

I think it should be "Escape" then.

> Also I got "WARNING: Setting context in section global doesn't exist"

Fixed it. Forgot to set the context setting to the right name.

**Comment by fwsmit** [comment]
I'm going to merge this. Thanks for testing @wgmayer0. If you still find a bug, let me know!

**Comment by wgmayer0** [comment]
10-4! Thanks for bringing back the shortcuts!

**Pull request merged**